src/hotspot/share/opto/library_call.cpp

```diff
@@ -5206,24 +5206,108 @@ bool LibraryCallKit::inline_vectorizedMismatch() {
   const TypeAryPtr* top_a = a_type->isa_aryptr();
   const TypeAryPtr* top_b = b_type->isa_aryptr();
   if (top_a == NULL || top_a->klass() == NULL ||
-    top_b == NULL || top_b->klass() == NULL) {
+      top_b == NULL || top_b->klass() == NULL) {
     // failed array check
     return false;
   }
-
-  Node* call;
   jvms()->set_should_reexecute(true);
 
   Node* obja_adr = make_unsafe_address(obja, aoffset);
   Node* objb_adr = make_unsafe_address(objb, boffset);
 
-  call = make_runtime_call(RC_LEAF,
-    OptoRuntime::vectorizedMismatch_Type(),
-    stubAddr, stubName, TypePtr::BOTTOM,
-    obja_adr, objb_adr, length, scale);
+  assert(scale->bottom_type()->isa_int() &&
+         scale->bottom_type()->is_int()->is_con(),
+         "non-constant scale value");
+
+  int scale_val = scale->bottom_type()->is_int()->get_con();
+  BasicType prim_types[] = {T_BYTE, T_SHORT, T_INT, T_LONG};
+  BasicType vec_basictype = prim_types[scale_val];
+  const Type* vec_type = Type::get_const_basic_type(vec_basictype);
+  int vec_len = UsePartialInlineSize / type2aelembytes(vec_basictype);
+
+  Node* length_in_bytes = _gvn.transform(new LShiftINode(length, scale));
+  Node* length_cmp = _gvn.transform(new CmpINode(length_in_bytes, intcon(UsePartialInlineSize)));
+  Node* cmp_res = _gvn.transform(new BoolNode(length_cmp, BoolTest::le));
+
+  // Enable partial in-lining if compare size is less than UsePartialInlineSize(default 32 bytes).
+  bool enable_pi = (UsePartialInlineSize > 32) ? (NULL != vec_type->isa_int())
```

Comment:
Can you elaborate on the check, please?

My reading is:
- `UsePartialInlineSize = 16: T_BYTE, T_SHORT` 
- `UsePartialInlineSize = 32: T_BYTE, T_SHORT` 
- `UsePartialInlineSize = 64: T_BYTE, T_SHORT, T_INT`

Why do you handle `T_LONG` case then?

What happens for `UsePartialInlineSize = 0`? Should partial inlining be just explicitly disabled in such case?